### PR TITLE
Initial CI touches

### PR DESCRIPTION
**Pull request by antoniovazquezblanco**
* Removes the need to check for target branch when installing sigutils as only master is used now.
* I've created packages for soapy in mingw so no need to build it in the pipeline.
* Removed an old TODO. Let me know if some of that should go to the issues section...

### CODE REVIEW

**Comment by antoniovazquezblanco** [comment]
Pipeline still fails for multiple reasons, I will get to that probably in future PRs.

**Comment by antoniovazquezblanco** [comment]
Magic 🪄

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -109,44 +73,16 @@ jobs:
...
-        sudo installer -pkg SDRplay_RSP_API-MacOSX-3.07.3.pkg -target /
-        # SoapySDRPlay3 from source
-        git clone https://github.com/pothosware/SoapySDRPlay3
-        cd SoapySDRPlay3
-        cmake -DCMAKE_BUILD_TYPE=Release -B build .
-        cmake --build build
-        cd build
-        sudo make install
-        cd ../..
+      run: brew install libsndfile volk fftw soapysdr
```
Comment (BatchDrake) [inline .github/workflows/build.yml:79]: Isn't json-c missing here?

**Comment by BatchDrake** [comment]
I am surprised that TODO was in the tree. Dang, that's from when Suscan was a GTK app

File: .github/workflows/build.yml (hunk 1 above)
Comment (antoniovazquezblanco) [reply .github/workflows/build.yml:79]: It compiles. Either something is wrong in cmake or it is not needed... I will check later...

Comment (BatchDrake) [reply .github/workflows/build.yml:79]: It compiles, but the resulting binary doesn't have SigMF support. The json-c dependency is optional but we need to test it here nonetheless.

Comment (antoniovazquezblanco) [reply .github/workflows/build.yml]: That was an easy fix :)

**Comment by antoniovazquezblanco** [comment]
Ready!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -187,14 +122,13 @@ jobs:
       - uses: msys2/setup-msys2@v2
         with:
           msystem: MINGW64
-          update: true
-          install: git mingw-w64-x86_64-cc mingw-w64-x86_64-make mingw-w64-x86_64-cmake mingw-w64-x86_64-libsndfile mingw-w64-x86_64-fftw mingw-w64-x86_64-volk mingw-w64-x86_64-libxml2 mingw-w64-x86_64-libusb
+          install: git mingw-w64-x86_64-cc mingw-w64-x86_64-make mingw-w64-x86_64-cmake mingw-w64-x86_64-libsndfile mingw-w64-x86_64-fftw mingw-w64-x86_64-volk mingw-w64-x86_64-libxml2 mingw-w64-x86_64-libusb mingw-w64-x86_64-soapysdr mingw-w64-x86_64-rtl-sdr mingw-w64-x86_64-soapyrtlsdr
```
Comment (BatchDrake) [inline .github/workflows/build.yml:125]: Does MinGW64 include json-c?

Comment (antoniovazquezblanco) [reply .github/workflows/build.yml:125]: Will do in a future PR :)

**Comment by antoniovazquezblanco** [comment]
Thanks!

**Pull request merged**